hazelcast/src/main/java/com/hazelcast/query/PagingPredicate.java

```diff
@@ -298,14 +274,46 @@ public Predicate getPredicate() {
 
     /**
      * Retrieve the anchor object which is the last value object on the previous page.
-     *
+     * <p/>
      * Note: This method will return `null` on the first page of the query result.
+     *
      * @return Map.Entry the anchor object which is the last value object on the previous page
      */
     public Map.Entry getAnchor() {
-        return anchorMap.get(page);
+        return anchorList.get(page).getValue();
     }
 
+    /**
+     * After each query, an anchor entry is set for that page.
+     * The anchor entry is the last entry of the query.
+     *
+     * @param anchor the last entry of the query
+     */
+    void setAnchor(int page, Map.Entry anchor) {
+        SimpleImmutableEntry anchorEntry = new SimpleImmutableEntry(page, anchor);
+        if (page < anchorList.size()) {
+            anchorList.set(page, anchorEntry);
+        } else {
+            anchorList.add(anchorEntry);
+        }
+    }
+
+    Map.Entry<Integer, Map.Entry> getNearestAnchorEntry() {
+        int size = anchorList.size();
```

Comment:
`anchorCount` or something like this would be better. otherwise it might be confused with `pageSize`